### ISSUE TITLE
Temp files are not removed after sync

### ISSUE THREAD

**Issue by gmaclennan** [digidem/osm-p2p-syncfile#18]
TL;DR need to clean up temp files after a successful or failed sync

Syncfile creates a temporary folder for the db in the folder passed in the second argument to the constructor. After successful sync this folder is never removed. Databases can grow over 50Mb, so with a user syncing a couple of times a day (syncing to another user, then syncing back) this folder could grow to a gigabyte in 10 days.

**Comment by noffle** [comment]
@gmaclennan How did you confirm this? From data in the field, or the tests?

I was just poking at the tests on `master` and realized that a few tests weren't calling `syncfile.close()`, which would explain multifeed databases being left behind.

cc @luandro

**Comment by noffle** [comment]
I pushed a branch that I think solves the test issues, but I don't know if Travis is actually running stuff anymore?

https://travis-ci.org/github/digidem/osm-p2p-syncfile/builds/756473601

**Comment by gmaclennan** [comment]
> @gmaclennan How did you confirm this? From data in the field, or the tests?
> 
Yes I realise this was not a very helpfully written bug report, and I will do better in the future, because now I cannot remember where this came from either. I vaguely remember it was seeing sync files being left in the temp folder, but nobody has been using USB sync for a while, so I'm not sure how much the issue continues to persist. Based on the date, I think this was when Aliya was in the field in Ecuador, and I think we saw it when she was trying to sync a few computers that were low on disk space. There is the persistent issue with sync, where if there is a sparse hypercore within the database then the sync will not complete successfully (it just hangs). My concern was that if the sync would just hang then the files could be left on disk without being removed.

**Comment by noffle** [comment]
> think this was when Aliya was in the field in Ecuador, and I think we saw it when she was trying to sync a few computers that were low on disk space. There is the persistent issue with sync, where if there is a sparse hypercore within the database then the sync will not complete successfully (it just hangs). My concern was that if the sync would just hang then the files could be left on disk without being removed.

Ok, I see two issues to investigate:

1. Syncfile temp data not getting cleaned up because disk space ran out + the sync was aborted.
2. Syncfile temp data not getting cleaned up because sync hanged and/or the app was unexpectedly closed.

I think these are both of a bigger scope than what we originally thought, and am wondering if this is still within a scope you feel comfortable with @luandro? These could be trickier more full-stack bugs to investigate & fix.

**Comment by luandro** [comment]
Rather not work on it at the moment, maybe later on, as I've learned a bit about the sync code.
